File: backend/db/test/mongodb.go
package db 

import (
	"context"
	"fmt"
	"log"

	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)

func connectToMongo() (*mongo.Client) {
	var err error
	clientOptions := options.Client().ApplyURI("mongodb://localhost:27017")
	client, err := mongo.Connect(context.TODO(), clientOptions)
	if err != nil {
		log.Fatal(err)
	}

	return client 
}

func main() {
	// 测试连接
	client := connectToMongo()
	err := client.Ping(context.TODO(), nil)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println("成功连接到 MongoDB!")
}

File: backend/db/mongodb.py
# Converted Python code for MongoDB connection

from pymongo import MongoClient
import logging

def connect_to_mongo():
    client_options = "mongodb://localhost:27017"
    try:
        client = MongoClient(client_options)
        return client
    except Exception as err:
        logging.error(err)
        raise


